torch/nn/modules/distance.py

```diff
@@ -1,6 +1,39 @@
-
+import torch
+from .module import Module
 # TODO: Cosine
-# TODO: CosineDistance - make sure lua's CosineDistance isn't actually cosine similarity
+
+
+class CosineDistance(Module):
+    """
+    Computes cosine similarity between two vectors x1,x2:
+
+          x1*x2
```

Comment:
You need to make it a monospace block to have it displayed correctly. I'd be for using the `:math:` tag and LaTeX.